File: rb101/RB101-RB109: Small Problems/Easy 7/multiplicative_average.rb
# mlutiplicative_average.rb
# Write a method that takes an Array of integers as input, multiplies all the
# numbers together, divides the result by the number of entries in the Array,
# and then prints the result rounded to 3 decimal places. Assume the array is
# non-empty.

# Pseudo-code:

# Data-Structure:
# input: an array of integers
# output: a multiplicative average of the numbers rounded to 3 decimal places.
# rules: multiplicative average = all numbers multiplied with each other and
# then divided by the total number of numbers.

def show_multiplicative_average(array)
  result = format("%.3f", array.reduce(:*).to_f / array.size)
  puts "The result is #{result}"
end

show_multiplicative_average([3, 5])
show_multiplicative_average([6])
show_multiplicative_average([2, 5, 7, 11, 13, 17])
puts '------'

# Launch School solution:

def show_multiplicative_average(numbers)
  product = 1.to_f
  numbers.each { |number| product *= number }
  average = product / numbers.size
  puts format('%.3f', average)
end

show_multiplicative_average([3, 5])
show_multiplicative_average([6])
show_multiplicative_average([2, 5, 7, 11, 13, 17])

File: js210/js210_small_problems/easy_4/multiplicative_average.js
function showMultiplicativeAverage(numbers) {
  let total = 1;

  for (let index = 0; index < numbers.length; index += 1) {
    total *= numbers[index];
  }

  let average = total / numbers.length;
  return average.toFixed(3);
}

console.log(showMultiplicativeAverage([3, 5]));
console.log(showMultiplicativeAverage([2, 5, 7, 11, 13, 17]));
